src/Symfony/Component/DependencyInjection/Compiler/ServiceLocatorTagPass.php

```diff
@@ -101,6 +92,31 @@ protected function processValue($value, bool $isRoot = false)
         return new Reference($id);
     }
 
+    /**
```

Comment:
In my view, this needs documentation to explain what it is doing and why. The method name certainly gives no indication as to what is going on here.

```suggestion
    /**
     * Replaces numeric service IDs with service names.
```

The method name could probably be improved along these lines, also, e.g. *replaceNumericServiceLocatorKeys*.

Reply:
Thanks for the suggestion!